woocommerce-gateway-stripe.php

```diff
@@ -485,6 +486,12 @@ protected function enable_upe( $settings ) {
 				$settings['upe_checkout_experience_accepted_payments'] = [];
 				$payment_gateways                                      = WC()->payment_gateways->payment_gateways();
 				foreach ( WC_Stripe_UPE_Payment_Gateway::UPE_AVAILABLE_METHODS as $method_class ) {
+					if ( WC_Stripe_UPE_Payment_Method_Link::class === $method_class ) {
+						// ENABLE UPE METHOD
+						$settings['upe_checkout_experience_accepted_payments'][] = $method_class::STRIPE_ID;
+						continue;
+					}
+
```

Comment:
I think we can reuse the logic from `disable_upe` and replace this with:

```php
if ( ! defined( "$method_class::LPM_GATEWAY_CLASS" ) ) {
	continue;
}
```

Then, you can remove [this line](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/4b627920bf9a03d7aa0af41cae9609519c1b6128/includes/payment-methods/class-wc-stripe-upe-payment-method-link.php#L13), so that the constant is not defined and allowed to be skipped here.

That way, [this](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2370/files#diff-e1e5e9b442c2b5d4105edbdc3787651333c1923dd623677025f72d25796900d6R518-R520) and [this other](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2370/files#diff-e1e5e9b442c2b5d4105edbdc3787651333c1923dd623677025f72d25796900d6R533-R536) changes wouldn't be necessary.

If we're not going to be allowing UPE, I think it makes more sense to not define any legacy payment method class and reuse the generic logic here.